### ISSUE TITLE
Run some `TestConfig` tests as scripts.

### ISSUE THREAD

**Issue by joshieDo** [foundry-rs/foundry#3961]
### Component

Forge

### Describe the feature you would like

Although `forge test` and `forge script` share many components, they do vary in their execution path. This leads to some issues (eg. https://github.com/foundry-rs/foundry/pull/3960) where the same cheatcode test passes on `forge test` but it would fail on `forge script`, and we don't know.

It would be nice if `TestConfig` could be configurable to additionally run some tests as scripts. Ideally we'd enable it for every cheatcode test.

**Comment by DaniPopes** [comment]
I don't think this applies anymore. Script was rewritten, and it uses the same components (Executor) as forge test.
